ramalama/daemon/service/command_factory.py

```diff
@@ -59,6 +59,9 @@ def _set_defaults(self):
         if "seed" not in self.request_args:
             self.request_args["seed"] = ""
 
+        if "max_tokens" not in self.request_args:
```

Comment:
After merging the [inference spec PR](https://github.com/containers/ramalama/pull/1959), no changes are required here (and apparently I forgot to remove this part in the daemon, will create a PR to do so soon).  

To make the max tokens accessible, it needs to be added to the context here:
https://github.com/containers/ramalama/blob/main/ramalama/command/context.py#L10
The new variable can then be used in the specs, for example llama.cpp:
https://github.com/containers/ramalama/blob/main/inference-spec/engines/llama.cpp.yaml
The new entry should be something like (based on the logic below):
```yaml
- name: "-n"
  description: "Maximum tokens to generate"
  value: "{{ args.max_tokens }}"
  if: "{{ args.max_tokens > 0 }}"
```

Reply:
The `perplexity` operation was not (yet) refactored to use the spec. So it either needs to be added and refactored or temporarily added to the perplexity code for now.